OrbitCore/LinuxPerfUtils.h

```diff
@@ -69,13 +136,29 @@ namespace LinuxPerfUtils
 
     perf_event_attr generic_perf_event_attr();
 
+    // perf_event_open for fork and exit.
     int32_t task_event_open(int32_t a_CPU);
 
+    // perf_event_open for context switches.
     int32_t context_switch_open(pid_t a_PID, int32_t a_CPU);
 
-    int32_t tracepoint_event_open(uint64_t a_TracepointID, pid_t a_PID, int32_t a_CPU, uint64_t additonal_sample_type = 0);
+    // perf_event_open for kernel tracepoints.
+    int32_t tracepoint_event_open(uint64_t a_TracepointID,
+                                  pid_t a_PID, int32_t a_CPU,
+                                  uint64_t additional_sample_type = 0);
+
+    // perf_event_open for stack sampling with frame-pointer unwinding by the kernel itself.
+    int32_t callchain_sample_event_open(pid_t a_PID, uint64_t a_Frequency);
+
+    // perf_event_open for stack sampling
+    int32_t stack_sample_event_open(pid_t a_PID, uint64_t a_Frequency);
 
+    // perf_event_open for uprobes.
     bool supports_perf_event_uprobes();
```

Comment:
Do we need this method? My question here is: Can we simply assume that uprobes exist? If not, and we need this call, do we also need similar methods for the other perf_event_open calls?

Reply:
This was done to condition the use of BpfTrace, I'll see what to do with it when we fully switch to dynamic instrumentation with perf_event_open.
We don't do the same for other perf_event_open calls because other perf_event_open features have been supported for a long time while uprobes is from 4.15 (at least in the headers).